kasa/smartdevice.py

```diff
@@ -205,15 +209,19 @@ def __init__(
         """
         self.host = host
         self.port = port
-
-        self.protocol = TPLinkSmartHomeProtocol(host, port=port)
+        self.protocol: TPLinkProtocol = TPLinkSmartHomeProtocol(host, port=port)
         self.credentials = credentials
         _LOGGER.debug("Initializing %s of type %s", self.host, type(self))
         self._device_type = DeviceType.Unknown
         # TODO: typing Any is just as using Optional[Dict] would require separate checks in
         #       accessors. the @updated_required decorator does not ensure mypy that these
         #       are not accessed incorrectly.
         self._last_update: Any = None
+
+        # Homeassistant uses the mac for storing devices and alias for display so this property allows updating of
+        # some properties from the new discovery info even if the device can't authenticate
+        self._requires_update_overrides: dict = {}
```

Comment:
Just a quick note, this is something that is currently done (partially, at least) using the `update_from_discover_info` method.